src/lazy/reader.rs

```diff
@@ -53,42 +55,46 @@ use crate::{IonError, IonResult};
 ///# Ok(())
 ///# }
 /// ```
-pub struct LazyReader<'data> {
-    system_reader: LazySystemReader<'data>,
+pub struct LazyReader<'data, F: LazyFormat<'data>> {
+    system_reader: LazySystemReader<'data, F>,
 }
 
-impl<'data> LazyReader<'data> {
-    pub fn new(ion_data: &'data [u8]) -> IonResult<LazyReader<'data>> {
```

Comment:
🗺️ As `LazyReader` is now generic over a `LazyFormat`, the `new()` method now lives in the concretely-typed `LazyBinaryReader` below.